turbo/consumers.py

```diff
@@ -31,62 +28,37 @@ def notify(self, event, *args, **kwargs):
         else:
             dom_target = f"{model._meta.verbose_name.lower()}_{pk}"
 
-        for request_id in self.requests[event["channel_name"]]:
-            instance = model.objects.get(pk=pk)
-            app, model_name = model_label.lower().split(".")
-            template_context = {
-                "action": action,
-                "dom_target": dom_target,
-            }
-            # Remove actions don't have contents, so only add context for model
-            # template if it's not a remove action.
-            if action != REMOVE:
-                template_context.update(
-                    {
-                        "object": instance,
-                        model_name.lower(): instance,
-                        "model_template": template,
-                        **extra_context,
-                    }
-                )
-
-            self.send_json(
+        instance = model.objects.get(pk=pk)
+        app, model_name = model_label.lower().split(".")
+        template_context = {
+            "action": action,
+            "dom_target": dom_target,
+        }
+        # Remove actions don't have contents, so only add context for model
+        # template if it's not a remove action.
+        if action != REMOVE:
+            template_context.update(
                 {
-                    "request_id": request_id,
-                    "data": render_to_string("turbo/stream.html", template_context),
+                    "object": instance,
+                    model_name.lower(): instance,
+                    "model_template": template,
+                    **extra_context,
                 }
             )
+        signed_channel_name = signer.sign(event["channel_name"])
+        self.send_json(
+            {
+                "signed_channel_name": signed_channel_name,
+                "data": render_to_string("turbo/stream.html", template_context),
+            }
+        )
 
     def receive_json(self, content, **kwargs):
-        signer = Signer()
-        request_id = content.get("request_id")
-        if request_id is None:
-            raise TurboStreamException("No request_id in subscription request.")
-        message_type = content.get("type")
+        channel_name = signer.unsign(content["signed_channel_name"])
```

Comment:
Do we want to catch the `BadSignature` and rethrow `TurboStreamException` again?

Reply:
I can add them back, I was just trying to get to the core of the bug I found. Thanks for reviewing!

Reply:
OK, added back!